cluster/engine.go

```diff
@@ -53,6 +55,14 @@ type Engine struct {
 	eventHandler    EventHandler
 	healthy         bool
 	overcommitRatio int64
+	ScheduleQueue   *list.List
+}
+
+//ScheduledItem FIX ME : works for affinities as it needs the conatainer name and Image
+//Need to add more types to support all filters
+type ScheduledItem struct {
```

Comment:
I believe the struct `Node` in `scheduler/node` is what you want. (okay it's a big bigger, but it's our "ScheduledItem")